docs/docs/02-concepts/01-preflight-and-inflight.md

```diff
@@ -334,6 +334,83 @@ Within the first clause of the `lift` block, a list of qualifications on preflig
 Statements within a `lift` block are exempt from the compiler's analyzer that tries to determine preflight object usage automatically.
 If an inflight method is directly or indirectly called within a `lift` block without sufficient resource qualifications, it may result in errors at runtime.
 
+## Building liftable abstractions
+
+Compute environments where inflight code is executed are called **inflight hosts**.
+For example, AWS Lambda, Fly.io machines, and Fargate containers are all examples of compute environments that can run inflight code.
+
+In order to run a piece of inflight code, the inflight host may need to be set up in a certain way.
+For example, to run an inflight function that reads from a storage bucket, the inflight host may require permissions to read from the bucket, or may require certain environment variables to be set, or may require network policies must be configured.
+
+Preflight classes can be used to encapsulate these requirements.
+Any preflight class can implement a method named `onLift` that is called when the class is used in inflight code, where requirements can be added to the inflight host.
+
+Here's an example where a class named `Model` requires an AWS Lambda function to have permission to invoke a model inference:
+
+```js playground example
+bring aws;
+
+pub class Model {
+  modelId: str;
+  new(modelId: str) {
+    this.modelId = modelId;
+  }
+
+  pub inflight invoke(body: Json): Json {
+    // call the AWS SDK to invoke the model...
+    return "success";
+  }
+
+  pub inflight printModelId() {
+    log("Model ID: {this.modelId}");
+  }
+
+  pub onLift(host: std.IInflightHost, ops: Array<str>) {
+    if ops.contains("invoke") {
+      if let lambda = aws.Function.from(host) {
+        lambda.addPolicyStatements({
+          actions: ["bedrock:InvokeModel"],
+          effect: aws.Effect.ALLOW,
+          resources: [
+            "arn:aws:bedrock:*::foundation-model/{this.modelId}"
+          ],
+        });
+      } else {
+        throw "Unsupported inflight host type";
+      }
+    } else {
+      // no requirements for other operations
+    }
+  }
+}
+```
+
+Here, `Model` has one public inflight method.
```

Comment:
Maybe I'm misunderstanding does it have one or two? `invoke` and `printModelId`?

Reply:
Oops, you're right. Fixed